go/vumitools/middleware.py

```diff
@@ -603,6 +604,54 @@ def handle_failure(self, failure, connector_name):
         return failure
 
 
+class ConversationMetricsMiddlewareConfig(MetricsMiddlewareConfig):
+
+    redis_manager = ConfigDict(
+        "Redis configuration parameters", default={}, static=True)
+
+
+class ConversationMetricsMiddleware(MetricsMiddleware):
+    """
+    Middleware that stores which conversations have received or sent messages
+
+    :param dict redis_manager:
+        Connection configuration details for Redis.
+    """
+
+    CONFIG_CLASS = ConversationMetricsMiddlewareConfig
+
+    @inlineCallbacks
+    def setup_middleware(self):
+        # We don't use a VumiApi here because we don't have a Riak config for
+        # it.
+        self.redis = yield TxRedisManager.from_config(
+            self.config.redis_manager)
+
+    def teardown_middleware(self):
+        return self.redis.close_manager()
+
+    @inlineCallbacks
+    def get_conv_key(self, msg):
+        mdh = MessageMetadataHelper(
+            self.vumi_api, msg, conversation_cache=self._conversation_cache)
```

Comment:
We don't retrieve the conversation, we just use its key, so we can remove `conversation_cache=self._conversation_cache` here.